content/blog/2021/07/2021-07-30-remoting-monitoring-phase-1.adoc

```diff
@@ -0,0 +1,216 @@
+
+|system.memory.utilization|1||
+|
+System memory utilization,
+see `com.sun.management.OperatingSystemMXBean.getTotalPhysicalMemorySize`
+and `com.sun.management.OperatingSystemMXBean.getFreePhysicalMemorySize`.
+Report 0% if no physical memory is discovered by the JVM.
+
+|system.paging.usage|bytes|state|`used`, `free`
+|
+see `com.sun.management.OperatingSystemMXBean.getFreeSwapSpaceSize`
+and `com.sun.management.OperatingSystemMXBean.getTotalSwapSpaceSize`.
+
+|system.paging.utilization|1||
+|
+see `com.sun.management.OperatingSystemMXBean.getFreeSwapSpaceSize`
+and `com.sun.management.OperatingSystemMXBean.getTotalSwapSpaceSize`.
+Report 0% if no swap memory is discovered by the JVM.
+
+|process.cpu.load|%||
+|Process CPU load. See `com.sun.management.OperatingSystemMXBean.getProcessCpuLoad`.
+
+|process.cpu.time|ns||
+|Process CPU time. See `com.sun.management.OperatingSystemMXBean.getProcessCpuTime`.
+
+.2+|runtime.jvm.memory.area .2+|bytes|type|`used`, `committed`, `max`
+.2+|see link:https://docs.oracle.com/en/java/javase/11/docs/api/java.management/java/lang/management/MemoryUsage.html[MemoryUsage]
+|area|`heap`, `non_heap`
+
+.2+|runtime.jvm.memory.pool .2+|bytes|type|`used`, `committed`, `max`
+.2+|see link:https://docs.oracle.com/en/java/javase/11/docs/api/java.management/java/lang/management/MemoryUsage.html[MemoryUsage]
+|pool|`PS Eden Space`, `G1 Old Gen`...
+
+|runtime.jvm.gc.time|ms|gc| `G1 Young Generation`, `G1 Old Generation`, ...
+|see link:https://docs.oracle.com/en/java/javase/11/docs/api/jdk.management/com/sun/management/GarbageCollectorMXBean.html[GarbageCollectorMXBean]
+
+|runtime.jvm.gc.count|1|gc| `G1 Young Generation`, `G1 Old Generation`, ...
+|see link:https://docs.oracle.com/en/java/javase/11/docs/api/jdk.management/com/sun/management/GarbageCollectorMXBean.html[GarbageCollectorMXBean]
+
+|===
+
+==== Traces
+
+We tried several approaches to instrument remoting module, but good approach is not established yet.
+
+Here is a draft documentation of the spans to collect. link:https://docs.google.com/document/d/1gjRamLWz3NwenVifC5pYyBMmxsUjl9MjspZF0mRYeaI/edit#heading=h.6xn68iwvd7gz[Google Doc]
+
+==== Logs
+
+Coming soon!
+
+=== Metric and span demo visualization
+
+Our team created a demo example with Docker compose and visualized the metrics and spans.
+
+*_Click to open in new tab_*
+
+image:../../../images/post-images/2021-07-30-remoting-monitoring-phase-1/prometheus-metrics.png[prometheus metric visualization, width=40%, link=/images/post-images/2021-07-30-remoting-monitoring-phase-1/prometheus-metrics.png, window=_blank]
+image:../../../images/post-images/2021-07-30-remoting-monitoring-phase-1/jaeger-spans.png[jaeger span visualization, width=55%, link=/images/post-images/2021-07-30-remoting-monitoring-phase-1/jaeger-spans.png, window=_blank]
+
+== Google Summer of Code Midterm Demo
+
+.Our project demo starts with 8:20
+video::_D0hiA1Cgz8[youtube,width=400,height=300]
```

Comment:
The `start=` markup will skip the 8 minutes of introduction and allow the viewer to immediately see the start of the presentation.  Optional but would be nice to have
```suggestion
video::_D0hiA1Cgz8[youtube,width=400,height=300,start=514]
```